packages/ckeditor5-engine/src/controller/datacontroller.ts

```diff
@@ -80,8 +105,8 @@ export default class DataController {
 			mapper: this.mapper,
 			schema: model.schema
 		} );
-		this.downcastDispatcher.on( 'insert:$text', insertText(), { priority: 'lowest' } );
-		this.downcastDispatcher.on( 'insert', insertAttributesAndChildren(), { priority: 'lowest' } );
+		this.downcastDispatcher.on<InsertEvent<ModelText | ModelTextProxy>>( 'insert:$text', insertText(), { priority: 'lowest' } );
```

Comment:
Maybe we should prefix those event types to indicate distinction between upcast and downcast events?